packages/core/src/storage/vectorStore/MilvusVectorStore.ts

```diff
@@ -103,36 +104,35 @@ export class MilvusVectorStore implements VectorStore {
       ],
     });
     await this.milvusClient.createIndex({
-      collection_name: this.collection,
+      collection_name: this.collectionName,
       field_name: this.embeddingKey,
     });
   }
 
-  public async connect(): Promise<void> {
-    await this.milvusClient.connectPromise;
+  private async ensureCollection(): Promise<void> {
+    if (!this.collection) {
+      await this.milvusClient.connectPromise;
+
+      // Check collection exists
+      const isCollectionExist = await this.milvusClient.hasCollection({
+        collection_name: this.collectionName,
+      });
+      if (!isCollectionExist.value) {
+        await this.createCollection();
+      }
 
-    // Check collection exists
-    const isCollectionExist = await this.milvusClient.hasCollection({
-      collection_name: this.collection,
-    });
-    if (!isCollectionExist.value) {
-      await this.createCollection();
+      await this.milvusClient.loadCollectionSync({
+        collection_name: this.collectionName,
+      });
+      this.collection = this.collectionName;
```

Comment:
it's a bit confusing that `this.collection` is a `string` - doesn't a `collectionInitialized` work?